### ISSUE TITLE
Juce migration

### ISSUE THREAD

**Issue by lucianodato** [lucianodato/noise-repellent#27]
For a gui and better functionalitty features

**Comment by simonvanderveldt** [comment]
@lucianodato Thanks for the plugin!
If you're still considering migrating to a framework I'd like to suggest using disthro.
JUCE is a nightmare, especially on Linux and it doesn't have LV2 support.

**Comment by lucianodato** [comment]
Thanks! I do have a branche where I've started that long ago

**Comment by lucianodato** [comment]
I got back to reading about the convenience of going juce or not. I think I might go with juce. I remember starting with the dpf migration but honestly I prefer juce given the documentation and that I have limited skills in plugin development. Any opinions?

**Comment by simonvanderveldt** [comment]
@lucianodato I'd be surprised if JUCE is really easier, it does have a lot more magic though :P
@falkTX did a talk about DPF quiet some time ago https://www.youtube.com/watch?v=FkHDs3Y18VY and he maintains a fork of JUCE that supports LV2. He might have a suggestion or two.

**Comment by lucianodato** [comment]
Yeah I saw this thread https://github.com/juce-framework/JUCE/issues/123. There are some hopes about it being integrated in the near future. Probably falktx knows more. Thanks for your response.

**Comment by falkTX** [comment]
For plugins on the more simple side JUCE is quite bloated.
DPF has VST3 support in the works, API is just as simple and supports LV2 from the get-go.
Oh and DPF is open to pull requests, something that JUCE is not (so you just hope they care enough about your specific issues)

PS: I hate that talk.. was not in the best mood that day :sweat_smile:

**Comment by falkTX** [comment]
and related to documentation, that is going to be a target for October.
with my employer (MOD Devices) we need to focus on plugin documentation to make the platform appealing to new devs, and DPF is the chosen framework for those that do not want to do raw LV2. (JUCE LV2 support is subpar, due it always been ignored for so many years, and the official one is likely still months away).

**Comment by pms967** [comment]
I'd say JUCE over DISTHRO. Though you may want to have a look also at this new one:

https://dplug.org/

**Comment by falkTX** [comment]
dplug requires rewriting in a new language, that is likely out of scope.

note though you can use the DPF graphics stuff with the current LV2 plugin, no need to rework the entire thing if all you want is a custom UI. I can help to set this up if it is wanted, as it is not a typical usecase.

**Comment by falkTX** [comment]
Sorry to bump so much, I was in the middle of writing something that is relevant here, a table of supported plugin features within DPF.
https://github.com/DISTRHO/DPF/blob/develop/FEATURES.md

Makes it much clearer what is available.

**Comment by lucianodato** [comment]
Hey falktx! Thanks for the clarifications. I'll keep looking at the possibilities. Thing is that I'm on a windows machine now and don't really know how to start with dpf on windows. That's my lazy self though, I'll look into it for sure!

**Comment by falkTX** [comment]
DPF supports cmake too, should be as simple as this https://github.com/DISTRHO/DPF/blob/main/examples/Parameters/CMakeLists.txt

but anyway, not trying to push you, use whatever works best.
though as someone else mentioned here on the beginning:

> JUCE is a nightmare, especially on Linux

There are pros and cons, evaluate carefully and all that. JUCE for sure has the biggest community around it, for being more popular. The company behind iLok owns them now though, something perhaps worth considering.

**Comment by lucianodato** [comment]
Thanks for your input! Yeah you might be right on this one... I'll stay open. Need to weight all the possibilities yet. And, big one, read the code again to remember it lol!

**Comment by lucianodato** [comment]
Finally I'm working on this. Here are some screenshots of what I did

<img width="1092" height="745" alt="Captura de pantalla 2026-07-31 a las 19 47 49" src="https://github.com/user-attachments/assets/22f688e9-2036-4ac8-aa8f-1417b40d01de" />

<img width="1319" height="948" alt="image" src="https://github.com/user-attachments/assets/eca16498-7c34-49b6-b486-7808d4d0b027" />

<img width="1092" height="745" alt="image" src="https://github.com/user-attachments/assets/1d237ac9-8d36-459c-85d1-db467d51f6d8" />

Pretty similar to most tools of the like
